airflow/providers/amazon/aws/transfers/dynamodb_to_s3.py

```diff
@@ -148,29 +174,58 @@ def hook(self):
         return DynamoDBHook(aws_conn_id=self.source_aws_conn_id)
 
     def execute(self, context: Context) -> None:
-        if self.export_time:
+        # There are 2 separate export to point in time configuration:
+        # 1. Full export, which takes the export_time arg.
+        # 2. Incremental export, which takes the incremental_export_... args
+        # Hence export time could not be used as the proper indicator for the `_export_table_to_point_in_time`
+        # function. This change introduces a new boolean, as the indicator for whether the operator scans
+        # and export entire data or using the point in time functionality.
+        if self.point_in_time_export:
```

Comment:
This change is breaking change. By introducing this change, you'll break DAGs of users using this operator. If a user uses this operator with `export_time` provided, you'll break their experience. Also, I would suggesting making this change in a different way. Why dont you expose a new parameter on the operator level `export_table_to_point_in_time_kwargs` which let the user define whatever extra parameter they want to `export_table_to_point_in_time` call? As a user you would be able to achieve what you are trying to do and the change would be way simpler

Reply:
I agree that it is a breaking change, but I would argue that it is a necessary one. 
1.  The point in time export has 2 separate types: `FULL_EXPORT` and `INCREMENTAL_EXPORT`. Only `FULL_EXPORT` is using the `export_time` argument, since it will trigger the full export till that point in time. For `INCREMENTAL_EXPORT` types the `export_time` is never used, instead, you would need to define a specific period where you want to export the data.  So I think the current `export_time` is not a good candidate for deciding if the operator should use point int time export. If I pass all the required params for incremental export to the `export_table_to_point_in_time` call, it will never run with `export_time` as the deciding factor. 

2. I think `export_table_to_point_in_time_kwargs` would have the same issue, no? If I pass all the parameter I want for incremental export, it would still not run it with` export_table_to_point_in_time` since the export_time is missing. 


I like the `export_table_to_point_in_time_kwargs`, but we would still need to solve the first issue above. 
Or am I missing something?  🤔

Reply:
Alternatively I can code an extra check in `_export_table_to_point_in_time` function to determine if the export_type is `FULL_EXPORT` or `INCREMENTAL_EXPORT `. And if it is `INCREMENTAL_EXPORT` I would set the `export_time` back to None to allow it be removed.   

I would solve the breaking change, but it also feels very backwards, meaning that we have to pass an `export_time` even if we want to do an incremental export,  and that value is never used.

Reply:
I see. What about modifying this if to:

```
if self.point_in_time_export or self.export_time:
```

If `self.export_time` then it means you want to use `_export_table_to_point_in_time`?

Reply:
Yes. That would work! 

I will make that change now.  😄

Reply:
made the change now. thanks for the suggestion. 😁

Reply:
Now we solved the backward compatibility thing, what do you think of my proposal using `export_table_to_point_in_time_kwargs` instead of adding X new parameters on the operator side?

Reply:
I like it. think it is also more future proof that way. 
But I need to test it locally first

Reply:
I have updated it now, please have a look when you have time :)